docs/ilab-model-engine.md

```diff
@@ -0,0 +1,201 @@
+|       |____chat
+|       |____evaluate
+|       |
+|_______data
+|       |
+|       |____generate
+|       |
+|_______config
+|       |
+|       |____init
+|       |
+|_______tax
+|       |
+|       |____diff
+|       |____check
+|       |____download
+```
+
+The main point of this structure is to introduce a group hierarchy and to improve the usability of the existing `ilab` structure. Some key things missing in `ilab` currently are:
+
+1. `ilab init` does too many things.
+    - We don't want "black box" commands. This creates bad UX and also ties too many processes together in which one might eventually need to be changed.
+2. ambiguity in "what" we are generating, training etc.
+    - The current structure of `ilab` requires an in depth knowledge of the ordering of commands and background knowledge for how our CLI is built. This is not the best way to grow. Commands that are not intuitive and require a deep dive into the docs just to get started might not engage users as we hope to. Having an intuitive structure in which people can see easily from the docstring what each commands does and how they might use them, makes more sense. We are generating data, then training a model, running a server to interact with that model, then finally chatting with the model
+3. Some commands like `ilab test` and `ilab list` should be renamed to show their true purpose (`ilab model evaluate`and `ilab taxonomy diff`)
+4. We will hit a point where we want to add more commands to `ilab` due to expected integration with the backend, model registries etc. This list will grow to a point where this form of top group organization is not maintainable.
+
+In general, moving to this structure allows for more growth when `ilab` reaches a stable release and encourages broader usage of the tool for all sorts of Open source AI models.
+
+Outside the scope of this EP, but still important in the near future, the following structure is preferred:
+
+```console
+ilab
+|
+|_______model
+|       |
+|       |____quantize *
+|       |____convert
+|       |____download
+|       |____train (--convert)
+|       |____top *
+|       |____serve   (-i)
+|       |____chat
+|       |____evaluate
+|       |____push *
+|       |____rm *
+|       |____ls *
+|       |____tag *
+|       |
+|_______data
+|       |____generate
+|       |____mix *
+|       |
+|_______config
+|       |
+|       |____create *
+|       |____edit *
+|       |____rm *
+|       |
+|_______tax
+|       |
+|       |____diff
+|       |____check
+|       |____download
+```
+
+The starred commands are new, and will be dealt with in another enhancement.
+
+#### Goals for 5/30/2024 milestone release of CLI
+
+For the next few milestones, it has been identified that the overall structure shift should be in place. We should aim to have the base hierarchy of commands in place that will exist for the future of `ilab` so that when users become familiar with the CLI, it does not change radically soon after launch.
```

Comment:
```suggestion
For the next few milestones, it has been identified that the overall structure shift should be in place. We should aim to have the base hierarchy of commands in place that will exist for the future of `ilab` so that when users become familiar with the CLI, further drastic changes are not needed.
```

since the project has already launched

Reply:
Unless you mean 1.0 launch @cdoern?

Reply:
no, this is an old sentence (this EP has been around for a while :) )